united.py

```diff
@@ -28,10 +54,224 @@ def parse_date(created):
+        accident_date[3] -= 3
+
+    # if accident happend after the last friday of march at 2:00 o'clock
+    elif (accident_date[1] == 3 & summer_clock[2] < accident_date[2] | (
+                    summer_clock[2] == accident_date[2] & accident_date[3] >= 2)):
+        accident_date[3] -= 3
+    else:  # winter_clock
+        accident_date[3] -= 2
+    # ]
+    adate = ''.join((accident_date.year, accident_date.month, accident_date.day, accident_date.hour))
+
+    i = -1
+
+    doc = Document()
+    base = doc.createElement('accident_date')
+    doc.appendChild(base)
+
+    for station in station_data_in_date:
+        i += 1
+        if adate in str(station.childNodes[i].nodeValue):
+            base.appendChild(get_parent_object_node(station))
+    print doc.toprettyxml(indent="    ", encoding="utf-8")
+
+
+
+def find_station_by_coordinate(latitude, longitude):
+
+    station_place_in_xml = -1
+    lat_difference = 0
+    lon_difference = 0
+    min_distance = 100000  # initialize big starting value so the distance will always be smaller than the initial
+    station_data = collection.getElementsByTagName('surface_station')
+    i = -1
+
+    for station in station_data:
+        i += 1  # save the place of the station in the file
+        station_lon = station.getElementsByTagName('station_lon')
+        for lon in station_lon:
+            lon_difference = math.pow(float(lon.childNodes[0].nodeValue) - float(longitude), 2)
+        station_lat = station.getElementsByTagName('station_lat')
+        for lat in station_lat:
+            lat_difference = math.pow(float(lat.childNodes[0].nodeValue) - float(latitude), 2)
+        temp_dis = math.sqrt(lat_difference + lon_difference)
+        if temp_dis < min_distance:
+            min_distance = temp_dis
+            station_place_in_xml = i
+
+    return station_place_in_xml
+
+
+def convert_xml_values_to_numbers(rain):
+    num_conv = rain[:2]  # variable to help convert from string to number
+
+    for char in num_conv:  # in the xml number are in a three digits format (4-004), we delete the 0es before the number
+        if char == '0':
+            rain.replace(char, '')
+        else:
+            break
+
+    rain_in_millimeters = float(rain)
+
+    if rain_in_millimeters >= 990:
+        # numbers that are higher then 990 in the xml code equals 0.(the last digit) for example 991 = 0.1
+        rain_in_millimeters *= 0.01
+        # str_rain = "0." + rain
+        # rain_in_millimeters = float(str_rain)  # convert string to numbers
+
+    return rain_in_millimeters
+
+
+def convert_xml_numbers_to_hours(rain_duration):
+    # convert the xml code for number of hours, to the actual number of hours
+    str_duration = str(rain_duration)
+    str_duration = str_duration.strip(' ')
+    hours = {"1": 6, "2": 12, "3": 18, "4": 24, "/": 24, "5": 1, "6": 2, "7": 3, "8": 9, "9": 15}
```

Comment:
extract what?

2016-04-17 11:06 GMT+03:00 Daniel Hershcovich <notifications@github.com>:

> In united.py
> <https://github.com/hasadna/anyway/pull/579#discussion_r59979982>:
>
> > +    rain_in_millimeters = float(rain)
> > +
> > +    if rain_in_millimeters >= 990:
> > +        # numbers that are higher then 990 in the xml code equals 0.(the last digit) for example 991 = 0.1
> > +        rain_in_millimeters *= 0.01
> > +        # str_rain = "0." + rain
> > +        # rain_in_millimeters = float(str_rain)  # convert string to numbers
> > +
> > +    return rain_in_millimeters
> > +
> > +
> > +def convert_xml_numbers_to_hours(rain_duration):
> > +    # convert the xml code for number of hours, to the actual number of hours
> > +    str_duration = str(rain_duration)
> > +    str_duration = str_duration.strip(' ')
> > +    hours = {"1": 6, "2": 12, "3": 18, "4": 24, "/": 24, "5": 1, "6": 2, "7": 3, "8": 9, "9": 15}
>
> Perhaps extract this to a constant with a proper name
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/hasadna/anyway/pull/579/files/2dd7573bd5bab0315ca68dc10db70ec63fcc8e09#r59979982>
>